Automattic/wp-calypso
# Inline the PostCSS config file into the webpack config

This is a follow up to @alshakero's `postcss-custom-properties` optimization in #42829.

Here I'm trying to speed up the webpack build by inlining the PostCSS config file into the webpack config. The `postcss-loader` doesn't need to load the config file 800+ times, for each `*.scss` compilation, but it's created only once, as an in-memory object, when creating the webpack config.

I needed to add a new option, `postCssOptions`, to the `calypso-build/webpack/sass` loader, enhancing the existing `postCssConfig` option. The existing option only lets us specify a path and context for the config file, but the new options lets up specify the `plugins` directly, skipping the config-file-loading step completely.

This speeds up mainly the `fallback` production build, the only one that uses `postcss-custom-properties` plugin. Testing locally, the timings improved from 167 seconds before to 94 seconds after (`yarn run build-client-fallback` with production env variables, skipping the linting check).

I'm not sure if these timings improvements are for real -- maybe the `cache-loader` and other similar things skew the result? It's worth trying to reproduce the results by the reviewer.

**How to test:**
Verify that the fallback build output still contains fallback definitions for CSS theme colors, and that the `autoprefixer` did its work, too.
